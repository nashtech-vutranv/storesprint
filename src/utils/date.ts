import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
