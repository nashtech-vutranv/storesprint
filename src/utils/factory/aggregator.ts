import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
