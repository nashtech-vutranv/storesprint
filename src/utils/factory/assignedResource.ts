import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
