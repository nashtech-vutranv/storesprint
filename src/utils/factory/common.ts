import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
