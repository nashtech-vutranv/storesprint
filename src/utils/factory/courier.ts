import faker from '@faker-js/faker'
import {ICourierMapping} from '../../interface'

export const mockCourier = (): ICourierMapping => {
  return {
    id: faker.datatype.string(),
    createdAt: faker.datatype.string(),
    modifiedAt: faker.datatype.string(),
    mmsValue: faker.datatype.string(),
    version: faker.datatype.number(),
    mappingTarget: {
        id: faker.datatype.string(),
        mappingTypeId: faker.datatype.string(),
        marketplaceCode: faker.datatype.string(),
        marketplaceTypeId: faker.datatype.string(),
        marketplaceValue: faker.datatype.string(),
    },
    mappingType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    mappingTypeId: faker.datatype.string(),
    marketplaceType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    marketplaceTypeId: faker.datatype.string(),
    marketplaceValue: faker.datatype.string(),
  }
}

export const mockCouriers = (): ICourierMapping[] => {
    return Array.apply(null, Array(10)).map(() =>mockCourier()) 
}
