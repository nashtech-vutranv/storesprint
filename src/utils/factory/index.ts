export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
