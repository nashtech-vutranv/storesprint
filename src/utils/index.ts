export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
