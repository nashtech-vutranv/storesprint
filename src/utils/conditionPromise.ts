import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}