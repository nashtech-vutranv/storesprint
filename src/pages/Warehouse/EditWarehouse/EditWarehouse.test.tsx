import {render} from '@testing-library/react'
import '@testing-library/jest-dom'
import Router, {MemoryRouter} from 'react-router-dom'
import {mockOption} from '../../../utils/factory'
import {EditWarehouse} from '../'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: any) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {
          // do smth
        }),
      },
    }
  },
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}))

beforeEach(() => {
  window.history.pushState(
    {
      selectedOrganizations: [mockOption()],
      eventKey: '0',
      currentStatus: null
    },
    '',
    ''
  )
})

jest.spyOn(Router, 'useParams').mockReturnValue({id: 'id'})

const siteLocaleReturn = Promise.resolve({
  data: {
    id: '751917528603943754',
    createdAt: '2022-04-13T02:35:24.862052Z',
    modifiedAt: '2022-04-13T02:35:24.862052Z',
    version: 0,
    erpId: '12345',
    platformId: 'null',
    name: 'Warehoue Name',
    status: 'ACTIVE',
  },
})

jest.mock('../../../services/WarehouseService', () => {
  const instance = jest.fn().mockReturnValue({
    getWarehouseById: jest.fn().mockImplementation(() => siteLocaleReturn),
  })
  return instance
})

test('renders Edit Warehouse', async () => {
  const tree = render(
    <MemoryRouter>
      <EditWarehouse />
    </MemoryRouter>
  )

  expect(tree).toMatchSnapshot()
  expect(tree.getByText(/warehouse_edit_page_title/i)).toBeInTheDocument()
})
