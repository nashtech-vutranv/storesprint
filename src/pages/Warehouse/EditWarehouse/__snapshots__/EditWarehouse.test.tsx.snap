// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Edit Warehouse 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form form-warehouse mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            warehouse_edit_page_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <form
              autocomplete="on"
              class="form-layout"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="mb-2 row"
                >
                  <div
                    class="col"
                  >
                    <p
                      class="d-flex align-items-center m-0"
                    >
                      <i
                        class="mdi mdi-alert-circle fs-3 pe-1"
                      />
                      warehouse_edit_warning_message
                    </p>
                  </div>
                </div>
                
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="erpId"
                    >
                      warehouse_edit_erpId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="erpId"
                      maxlength="64"
                      name="erpId"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="name"
                    >
                      warehouse_edit_organization_name
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="name"
                      maxlength="128"
                      name="name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center row"
                >
                  <div
                    class="mt-3 col-12"
                  >
                    <label
                      class="fs-4 fw-bold"
                      for="additionalInformation"
                    >
                      warehouse_edit_warehouse_address
                      :
                    </label>
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="addressLine1"
                    >
                      warehouse_edit_warehouse_addressLine1
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="addressLine1"
                      maxlength="255"
                      name="addressLine1"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="addressLine2"
                    >
                      warehouse_edit_warehouse_addressLine2
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="addressLine2"
                      maxlength="255"
                      name="addressLine2"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="addressLine3"
                    >
                      warehouse_edit_warehouse_addressLine3
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="addressLine3"
                      maxlength="255"
                      name="addressLine3"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="country"
                    >
                      warehouse_edit_warehouse_country
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component p-1"
                      id="country"
                      maxlength="255"
                      name="country"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="isActive"
                    >
                      warehouse_edit_active
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <div
                      class="p-inputswitch p-component p-inputswitch-checked p-disabled"
                    >
                      <div
                        class="p-hidden-accessible"
                      >
                        <input
                          aria-checked="true"
                          aria-label="switch"
                          checked=""
                          disabled=""
                          name="status"
                          role="switch"
                          type="checkbox"
                        />
                      </div>
                      <span
                        class="p-inputswitch-slider"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-8 offset-4"
                  >
                    <button
                      class="me-2 btn btn-danger"
                      type="button"
                    >
                      common_confirm_cancel
                    </button>
                    <button
                      class="me-2 btn btn-success"
                      disabled=""
                      type="button"
                    >
                      common_confirm_save
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form form-warehouse mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          warehouse_edit_page_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <form
            autocomplete="on"
            class="form-layout"
          >
            <div
              class="container-fluid"
            >
              <div
                class="mb-2 row"
              >
                <div
                  class="col"
                >
                  <p
                    class="d-flex align-items-center m-0"
                  >
                    <i
                      class="mdi mdi-alert-circle fs-3 pe-1"
                    />
                    warehouse_edit_warning_message
                  </p>
                </div>
              </div>
              
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="erpId"
                  >
                    warehouse_edit_erpId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="erpId"
                    maxlength="64"
                    name="erpId"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="name"
                  >
                    warehouse_edit_organization_name
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="name"
                    maxlength="128"
                    name="name"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center row"
              >
                <div
                  class="mt-3 col-12"
                >
                  <label
                    class="fs-4 fw-bold"
                    for="additionalInformation"
                  >
                    warehouse_edit_warehouse_address
                    :
                  </label>
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="addressLine1"
                  >
                    warehouse_edit_warehouse_addressLine1
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="addressLine1"
                    maxlength="255"
                    name="addressLine1"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="addressLine2"
                  >
                    warehouse_edit_warehouse_addressLine2
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="addressLine2"
                    maxlength="255"
                    name="addressLine2"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="addressLine3"
                  >
                    warehouse_edit_warehouse_addressLine3
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="addressLine3"
                    maxlength="255"
                    name="addressLine3"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="country"
                  >
                    warehouse_edit_warehouse_country
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component p-1"
                    id="country"
                    maxlength="255"
                    name="country"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="isActive"
                  >
                    warehouse_edit_active
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <div
                    class="p-inputswitch p-component p-inputswitch-checked p-disabled"
                  >
                    <div
                      class="p-hidden-accessible"
                    >
                      <input
                        aria-checked="true"
                        aria-label="switch"
                        checked=""
                        disabled=""
                        name="status"
                        role="switch"
                        type="checkbox"
                      />
                    </div>
                    <span
                      class="p-inputswitch-slider"
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-8 offset-4"
                >
                  <button
                    class="me-2 btn btn-danger"
                    type="button"
                  >
                    common_confirm_cancel
                  </button>
                  <button
                    class="me-2 btn btn-success"
                    disabled=""
                    type="button"
                  >
                    common_confirm_save
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
